bindings/experimental/distrdf/python/DistRDF/Backends/Dask/Backend.py

```diff
@@ -74,6 +73,25 @@ def _(cluster: JobQueueCluster, client: Client) -> int:
                            "Please report this as a bug.") from e
 
 
+def get_total_cores(cluster: SpecCluster, client: Client) -> int:
+    """
+    Retrieve the total number of cores from a Dask cluster object.
+    """
+    try:
+        # It may happen that the user is connected to a batch system. We try
+        # to import the 'dask_jobqueue' module lazily to avoid a dependency.
+        from dask_jobqueue import JobQueueCluster
+    except ModuleNotFoundError:
+        # If the 'dask_jobqueue' module is not found we can safely use the
+        # generic function.
+        return get_total_cores_generic(client)
+
+    if isinstance(cluster, JobQueueCluster):
+        return get_total_cores_jobqueuecluster(cluster)
+    else:
+        return get_total_cores_generic(client)
```

Comment:
maybe simpler, but up to you:
```suggestion
    try:
        # It may happen that the user is connected to a batch system. We try
        # to import the 'dask_jobqueue' module lazily to avoid a dependency.
        from dask_jobqueue import JobQueueCluster
        if isinstance(cluster, JobQueueCluster):
            return get_total_cores_jobqueuecluster(cluster)
    except ModuleNotFoundError:
        # We are not using dask_jobqueue, fall through to generic case
        pass
    return get_total_cores_generic(client)
```